### PR TITLE
spirv-as: Assume target from spvasm text

### PR DESCRIPTION
**Pull request by dneto0**
Assume the desired SPIR-V target version when the command line option is
    not provided. This is especially important when the source being
    assembled does not match the default (latest version), because it may
    create an invalid SPIR-V binary result.
    
    Also, clarify the wording for where a copy of spirv-headers is expected
    (to match expected capitalization).

### CODE REVIEW

**Comment by dneto0** [comment]
Originally from #5888

File: source/spirv_target_env.cpp [hunk 1]
```diff
@@ -172,6 +144,46 @@ bool spvParseTargetEnv(const char* s, spv_target_env* env) {
...
+      }
+      // j will match the prefix length if all characters before matched
+      if (j == kPrefixLength) {
+        char minor = text[i + j];
+        if (minor >= '0' && minor <= '6' && !std::isdigit(text[i + j + 1])) {
+          auto offset = minor - '0';
+          *env = spvTargetEnvNameMap[kSpvEnvUniversalStart + offset].second;
+          return true;
+        }
+      }
```
[the remark this answers is not in the document]
Comment (alan-baker) [reply source/spirv_target_env.cpp:173]: Maybe just tie it to the headers used to build those tools? I think we already store the major and minor versions from the grammar, but it is also available as SPV_VERSION in spirv.h.

File: source/spirv_target_env.cpp [hunk 2]
```diff
@@ -128,32 +130,43 @@ uint32_t spvVersionForTargetEnv(spv_target_env env) {
...
-    {"opencl2.1embedded", SPV_ENV_OPENCL_EMBEDDED_2_1},
-    {"opencl2.1", SPV_ENV_OPENCL_2_1},
-    {"opencl2.2embedded", SPV_ENV_OPENCL_EMBEDDED_2_2},
-    {"opencl2.2", SPV_ENV_OPENCL_2_2},
-    {"opengl4.0", SPV_ENV_OPENGL_4_0},
-    {"opengl4.1", SPV_ENV_OPENGL_4_1},
-    {"opengl4.2", SPV_ENV_OPENGL_4_2},
-    {"opengl4.3", SPV_ENV_OPENGL_4_3},
-    {"opengl4.5", SPV_ENV_OPENGL_4_5},
+// When a new SPIR-V version is released, update this table.
```
Comment (dneto0) [inline source/spirv_target_env.cpp:133]: There was no reason to put this in the header, so i pulled it back in the .cpp file.

File: source/spirv_target_env.cpp [hunk 3]
```diff
@@ -155,31 +197,43 @@ bool spvReadEnvironmentFromText(std::vector<char>& text, spv_target_env* env) {
       // Try to match against the expected version string
       constexpr const char* kVersionPrefix = "; Version: 1.";
       constexpr const auto kPrefixLength = 13;
-      constexpr const auto kMinimumVersionLength = kPrefixLength + 2;
-      if (i + kMinimumVersionLength >= text.size()) return false;
+      // Where do we expect to find the minor version digit.
```
Comment (s-perron) [inline source/spirv_target_env.cpp:200]: Is this a question or a statement?

**Review by s-perron — APPROVED** [review]
Thanks for the changes.

File: source/spirv_target_env.cpp (hunk 3 above)
Comment (dneto0) [reply source/spirv_target_env.cpp:200]: heh. it's a statement

Comment (dneto0) [reply source/spirv_target_env.cpp:200]: Changed the comment to `'minor_digit_pos' is the expected position of the version digit. `

**Pull request merged**